File: Python/invertirCadena.py
'''
Reto de "INVIRTIENDO CADENAS"
Dificultad: FÁCIL

Enunciado: Crea un programa que invierta el orden de una cadena de texto sin usar funciones propias del lenguaje que lo hagan de forma automática.
- Si le pasamos "Hola mundo" nos retornaría "odnum aloH"
'''

def invertir_cadena(cadena):
    cadena_invertida = ""
    for caracter in reversed(cadena):
        cadena_invertida += caracter
    return cadena_invertida

# Ejemplo de uso
cadena_original = "Hola mundo"
cadena_invertida = invertir_cadena(cadena_original)

print(f"Cadena original: {cadena_original}")
print(f"Cadena invertida: {cadena_invertida}")

File: Kotlin/invertirCadena.kt
package dev.krispi.invertirCadena

/*
 * Reto de "INVIRTIENDO CADENAS"
 * Dificultad: FÁCIL
 *
 * Enunciado: Crea un programa que invierta el orden de una cadena de texto sin usar funciones propias del lenguaje que lo hagan de forma automática.
 * - Si le pasamos "Hola mundo" nos retornaría "odnum aloH"
 */

fun invertirCadena(cadena: String): String {
    var cadenaInvertida = ""
    for (i in cadena.length - 1 downTo 0) {
        cadenaInvertida += cadena[i]
    }
    return cadenaInvertida
}

fun main() {
    val texto = "Hola mundo"
    val textoInvertido = invertirCadena(texto)
    println("Texto original: $texto")
    println("Texto invertido: $textoInvertido")
}
